### PR TITLE
Extract Title in folio transformer

### PR DESCRIPTION
**Pull request by anubhavijay**
## What does this change?

Extracts the title field from MARC records in the Folio transformer. This adds title extraction to the FOLIO adapter's transformation pipeline by:
- Importing the existing `extract_title` function from the MARC transformer module
- Using the extracted title when creating `WorkData` during FOLIO record transformation
- Updating the minimal record BDD test to include a 245 MARC field (title field) with a test value
- Adding error handling to validate that empty title fields raise `ValueError`

## How to test

Run the Folio transformer tests:

```bash
cd catalogue_graph
uv run pytest tests/adapters/transformers/folio/ -v
```

The test suite includes:
- **Minimal record scenario**: Validates that the title field is properly extracted and available in the transformed work data
- **Empty title scenario**: Confirms that a MARC record with an empty 245 field raises `ValueError` as expected

## How can we measure success?

- FOLIO records now have titles extracted from the MARC 245 field
- Existing Folio transformer tests pass with the new title extraction in place
- Both happy path and error path are covered: records with valid titles extract correctly, and empty titles are rejected with appropriate errors
- BDD test scenarios confirm the transformer correctly processes records with title fields and handles edge cases

## Have we considered potential risks?

- The `extract_title` function is already tested and used in other MARC-based transformers (e.g., Sierra, METS)
- This change integrates existing, proven functionality into the Folio transformer
- No changes to external APIs or schemas
- The change is additive — only adds title data to records that didn't previously have it extracted
- Error handling for malformed/empty titles follows the same pattern as existing transformers

### CODE REVIEW

File: catalogue_graph/src/adapters/transformers/folio_transformer.py [hunk 1]
```diff
@@ -31,7 +32,9 @@ def extract_predecessor_id(self, marc_record: Record) -> str | None:
     def transform_record(
         self, marc_record: Record, source_modified_time: datetime
     ) -> VisibleSourceWork:
-        work_data = WorkData()
+        work_data = WorkData(
+            title=extract_title(marc_record),
```
[the remark this answers is not in the document]
Comment (anubhavijay) [reply catalogue_graph/src/adapters/transformers/folio_transformer.py:36]: Title should be a mandatory for the work similar to existing transformers

**Review by kenoir — APPROVED** [review]
Looks good, the other co-pilot comments need addressing though.

Have you tried running this with local ES to see it working?

File: catalogue_graph/tests/adapters/transformers/folio/BDD/conftest.py [hunk 2]
```diff
@@ -2,8 +2,9 @@
 
 from datetime import datetime
 
+import pytest
 from pymarc.record import Record
```
[the remark this answers is not in the document]
Comment (anubhavijay) [reply catalogue_graph/tests/adapters/transformers/folio/BDD/conftest.py:6]: Ran ruff locally to fix these and have pushed the changes

File: catalogue_graph/tests/adapters/transformers/folio/BDD/conftest.py [hunk 3]
```diff
@@ -22,3 +23,11 @@ def do_transform(marc_record: Record) -> VisibleSourceWork:
     return transformer.transform_record(
         marc_record, source_modified_time=datetime(2020, 1, 1)
     )
+
```
[the remark this answers is not in the document]
Comment (anubhavijay) [reply catalogue_graph/tests/adapters/transformers/folio/BDD/conftest.py:26]: Run ruff format to address this and then have pushed the code

**Comment by anubhavijay** [comment]
> Looks good, the other co-pilot comments need addressing though.
> 
> Have you tried running this with local ES to see it working?

yesI  have run this locally on ES and checked the titles from folio interface.

image
